drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,454 @@
+
+	nxp_irtc_unlock_registers(irtc_reg);
+	irtc_reg->SECONDS = RTC_SECONDS_SEC_CNT(timeptr->tm_sec);
+
+	irtc_reg->HOURMIN = RTC_HOURMIN_MIN_CNT(timeptr->tm_min) |
+		RTC_HOURMIN_HOUR_CNT(timeptr->tm_hour);
+
+	/* 1 is valid for rtc_time.tm_wday property but is out of bounds for IRTC registers */
+	irtc_reg->DAYS = RTC_DAYS_DAY_CNT(timeptr->tm_mday) |
+		      (timeptr->tm_wday == -1 ? 0 : RTC_DAYS_DOW(timeptr->tm_wday));
+
+	irtc_reg->YEARMON = RTC_YEARMON_MON_CNT(calc_month) | RTC_YEARMON_YROFST(calc_year);
+
+	if (timeptr->tm_isdst != -1) {
+		irtc_reg->CTRL |= RTC_CTRL_DST_EN(timeptr->tm_isdst);
+		data->is_dst_enabled = true;
+	}
+
+	irq_unlock(key);
+
+	return 0;
+}
+
+static int nxp_irtc_get_time(const struct device *dev, struct rtc_time *timeptr)
+{
+	const struct nxp_irtc_config *config = dev->config;
+	struct nxp_irtc_data *data = dev->data;
+	RTC_Type *irtc_reg = config->base;
+
+	if (!timeptr) {
+		return -ENODATA;
+	}
+
+	timeptr->tm_sec =
+		((irtc_reg->SECONDS & RTC_SECONDS_SEC_CNT_MASK) >> RTC_SECONDS_SEC_CNT_SHIFT);
+	timeptr->tm_min =
+		((irtc_reg->HOURMIN & RTC_HOURMIN_MIN_CNT_MASK) >> RTC_HOURMIN_MIN_CNT_SHIFT);
+	timeptr->tm_hour =
+		((irtc_reg->HOURMIN & RTC_HOURMIN_HOUR_CNT_MASK) >> RTC_HOURMIN_HOUR_CNT_SHIFT);
+	timeptr->tm_wday = ((irtc_reg->DAYS & RTC_DAYS_DOW_MASK) >> RTC_DAYS_DOW_SHIFT);
+	timeptr->tm_mday = ((irtc_reg->DAYS & RTC_DAYS_DAY_CNT_MASK) >> RTC_DAYS_DAY_CNT_SHIFT);
+	timeptr->tm_mon =
+		((irtc_reg->YEARMON & RTC_YEARMON_MON_CNT_MASK) >> RTC_YEARMON_MON_CNT_SHIFT) - 1;
+	timeptr->tm_year = (int8_t)((irtc_reg->YEARMON & RTC_YEARMON_YROFST_MASK) >>
+				    RTC_YEARMON_YROFST_SHIFT) +
+			   RTC_NXP_IRTC_YEAR_OFFSET;
+	if (data->is_dst_enabled) {
+		timeptr->tm_isdst =
+			((irtc_reg->CTRL & RTC_CTRL_DST_EN_MASK) >> RTC_CTRL_DST_EN_SHIFT);
+	}
+
+	/* There is no nano second support for IRTC */
+	timeptr->tm_nsec = 0;
+	/* There is no day of the year support for IRTC */
+	timeptr->tm_yday = -1;
+
+	return 0;
+}
+
+#if defined(CONFIG_RTC_ALARM) || defined(__DOXYGEN__)
+static int nxp_irtc_alarm_get_supported_fields(const struct device *dev, uint16_t id,
+					       uint16_t *mask)
+{
+	ARG_UNUSED(dev);
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	*mask = (RTC_ALARM_TIME_MASK_SECOND | RTC_ALARM_TIME_MASK_MINUTE |
+		 RTC_ALARM_TIME_MASK_HOUR | RTC_ALARM_TIME_MASK_MONTHDAY |
+		 RTC_ALARM_TIME_MASK_MONTH | RTC_ALARM_TIME_MASK_YEAR);
+
+	return 0;
+}
+
+static int nxp_irtc_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+				   const struct rtc_time *timeptr)
+{
+	const struct nxp_irtc_config *config = dev->config;
+	struct nxp_irtc_data *data = dev->data;
+	RTC_Type *irtc_reg = config->base;
+
+	if (id != 0 || (mask && (timeptr == 0)) ||
+	    (timeptr && !is_rtc_time_in_bounds(timeptr, mask))) {
+		return -EINVAL;
+	}
+
+	uint32_t key = irq_lock();
+
+	nxp_irtc_unlock_registers(irtc_reg);
+
+	if ((mask & RTC_ALARM_TIME_MASK_SECOND) == RTC_ALARM_TIME_MASK_SECOND) {
```

Comment:
You only need the first comparison:
```
if (mask & RTC_ALARM_TIME_MASK_SECOND) {
        ...
}
```

Reply:
Thank you, I will fix this on the next update